addons/sms/wizard/sms_composer.py

```diff
@@ -239,10 +239,13 @@ def _action_send_sms_mass(self, records=None):
         records = records if records is not None else self._get_records()
 
         sms_record_values = self._prepare_mass_sms_values(records)
-        sms_all = self._prepare_mass_sms(records, sms_record_values)
-        if sms_all and self.mass_keep_log and records and isinstance(records, self.pool['mail.thread']):
+        records = records.filtered(lambda r: r.id in sms_record_values)
```

Comment:
Should be able to just browse the keys actually
`records = self.env[records._name].browse(list(sms_record_values))`